### PR TITLE
UF-288: Support custom icons in the Dock tabs

### PR DESCRIPTION
**Pull request by paulovmr**

### CODE REVIEW

**Comment by ederign** [comment]
I'm on this.

File: uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/items/SideDockItemFocused.java [hunk 1]
```diff
@@ -53,13 +54,15 @@ public SideDockItemFocused( final SideDockItem parent ) {
...
         add( uiBinder.createAndBindUi( this ) );
-        itemButton.setIconPosition( IconPosition.LEFT );
-        itemButton.setIcon( parent.getIcon() );
-        itemButton.setIconFixedWidth( true );
+        removeStyleName( "gwt-PopupPanel" );
+        createButton( parent );
+    }
+
+    private void createButton( final SideDockItem parent ) {
+        parent.configureIcon( itemButton, parent.getDock().getImageIconFocused() );
```
Comment (ederign) [inline uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/items/SideDockItemFocused.java]: This shouldn't verify if it is a BS icon or a image?

**Comment by ederign** [comment]
Tests?

**Comment by ederign** [comment]
@paulovmr small question inline and also please add tests for it. Thanks.

File: uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/items/SideDockItemFocused.java (hunk 1 above)
Comment (paulovmr) [inline uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/items/SideDockItemFocused.java]: The verification is inside the configureIcon method.

**Comment by paulovmr** [comment]
@ederign Tests added.

**Comment by ederign** [comment]
@paulovmr looks good. thanks. just a minor css issue:  https://www.dropbox.com/s/h9agiseobthmudl/Screen%20Shot%202016-03-11%20at%2010.17.04%20AM.png?dl=0.

ping me when u fix it.

**Comment by paulovmr** [comment]
@ederign Fixed. Thanks for catching it!

**Comment by ederign** [comment]
@paulovmr fixed the css. +1

**Comment by csadilek** [comment]
@ederign @paulovmr should the optaplanner images really be stored as part of UberFire? ?Also why is the interface called UberfireWiresImageResources?

**Comment by ederign** [comment]
@csadilek I think is ok because that is  on your showcase (wires web app)

**Comment by csadilek** [comment]
@ederign yeah I see it's just for testing purposes, still not ideal. I guess we can clean up the showcases another time though. I will merge this later today.

**Comment by paulovmr** [comment]
Jenkins please retest this

**Comment by paulovmr** [comment]
Depends on: https://github.com/uberfire/uberfire/pull/313

**Comment by paulovmr** [comment]
Jenkins please retest this

**Comment by csadilek** [comment]
master: e214dde9db3f6c0c5dfbfb15af9b610588c16b83

**Pull request closed without merging**